integration/test/ipam/provider.go

```diff
@@ -0,0 +1,154 @@
+// +build k8srequired
+
+package ipam
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/aws-operator/integration/env"
+	"github.com/giantswarm/aws-operator/integration/setup"
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	"github.com/giantswarm/e2e-harness/pkg/release"
+	e2eclientsaws "github.com/giantswarm/e2eclients/aws"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ProviderConfig struct {
+	AWSClient *e2eclientsaws.Client
+	Host      *framework.Host
+	Logger    micrologger.Logger
+	Release   *release.Release
+}
+
+type Provider struct {
+	awsClient *e2eclientsaws.Client
+	host      *framework.Host
+	logger    micrologger.Logger
+	release   *release.Release
+}
+
+func NewProvider(config ProviderConfig) (*Provider, error) {
+	if config.AWSClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.AWSClient must not be empty", config)
+	}
+	if config.Host == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Host must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.Release == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Release must not be empty", config)
+	}
+
+	p := &Provider{
+		awsClient: config.AWSClient,
+		host:      config.Host,
+		logger:    config.Logger,
+		release:   config.Release,
+	}
+
+	return p, nil
+}
+
+func (p *Provider) CreateCluster(ctx context.Context, id string) error {
+	setupConfig, err := p.newSetupConfig(id)
```

Comment:
We have to construct the setup config with the tenant cluster framework for the specific ID. For now this might just be fine and in the future we can make this a bit nicer and have the primitives for tenant cluster setup and management a bit straightened out. Right now we just have a blob with everything which we dispatch everywhere and this could maybe be a bit nicer later.